beacon-chain/p2p/handshake.go

```diff
@@ -35,20 +35,20 @@ func (s *Service) AddConnectionHandler(reqFunc func(ctx context.Context, id peer
 				log := log.WithField("peer", conn.RemotePeer())
 				if conn.Stat().Direction == network.DirInbound {
 					log.Debug("Not sending hello for inbound connection")
-					return
-				}
-				log.Debug("Performing handshake with peer")
-				if err := reqFunc(ctx, conn.RemotePeer()); err != nil && err != io.EOF {
```

Comment:
We should perform handshakes with incoming connections. Why change this?

Reply:
Edit: We should perform / request handshakes on both inbound and outbound connections. i.e. all peers.

Reply:
The current code returns immediately if there is no inbound connection; this change just allows control to flow past the handshake piece.

As to why the code isn't doing a handshake for inbound connections: no idea, but if I had to guess it would be that the outbound peer always initiates the handshake so the inbound doesn't need to do so as well.

Reply:
Sounds like existing issue. I'll file a bug. thanks!